cmd/kata-pkgsync/cli/main.go

```diff
@@ -0,0 +1,274 @@
+	var projects []obsgo.Project
+
+	for n, p := range cfgProjects {
+		proj := obsgo.Project{
+			User:     p.Auth.User,
+			Password: p.Auth.Password,
+		}
+
+		if len(p.Archs) == 0 {
+			p.Archs = append(p.Archs, "")
+		}
+
+		if len(p.Releases) == 0 {
+			p.Releases = append(p.Releases, "")
+		}
+
+		// Kata projects names format is "project:release:architecture"
+		for _, arch := range p.Archs {
+			for _, release := range p.Releases {
+				var fullname strings.Builder
+
+				fullname.WriteString(n)
+				if arch != "" {
+					fmt.Fprintf(&fullname, ":%s", arch)
+				}
+				if release != "" {
+					fmt.Fprintf(&fullname, ":%s", release)
+				}
+
+				proj.Name = fullname.String()
+				projects = append(projects, proj)
+			}
+		}
+	}
+
+	return projects
+}
+
+func getXferBinaries(pkg obsgo.PackageInfo, pcDistro string,
+	pcPackages []pkgcloud.Package, pcPackagesNeeded *[]bool) []obsgo.PkgBinary {
+
+	var xferBins []obsgo.PkgBinary
+	for _, src := range pkg.Files {
+		found := false
+		for i, dst := range pcPackages {
+			logrus.Debugf("Checking pkg %s %s VS %s %s", src.Filename, pcDistro, dst.Filename, dst.DistroVersion)
+			if pcDistro == dst.DistroVersion && src.Filename == dst.Filename {
+				logrus.Debugf("%s already on Packagecloud %s", src.Filename, pcDistro)
+				found = true
+				(*pcPackagesNeeded)[i] = true
+				break
+			}
+		}
+		if !found {
+			logrus.Debugf("%s NOT yet on Packagecloud %s", src.Filename, pcDistro)
+			xferBins = append(xferBins, src)
+		}
+	}
+
+	return xferBins
+}
+
+func main() {
+	flag.Usage = usage
+	verbose := flag.Bool("debug", false, "debug mode")
+	dlPath := flag.String("dir", defaultOBSDest, "Destination directory of packages download from OBS")
+	dryRun := flag.Bool("dry-run", false, "dry-run mode (do not download/upload files)")
+	pcDelete := flag.Bool("delete", false, "Delete Packagecloud packages that are not published on OBS")
+	showVersion := flag.Bool("version", false, "show the version")
+
+	flag.Parse()
+
+	if *showVersion {
+		fmt.Printf("%s %s (commit %v)\n", name, version, commit)
+		os.Exit(0)
+	}
+
+	if *verbose {
+		fmt.Println("Starting in debug mode...")
+		logrus.SetLevel(logrus.DebugLevel)
+	}
+
+	var configFile string
+	if len(flag.Args()) > 0 {
+		configFile = flag.Args()[0]
+	} else {
+		configFile = defaultConfig
+	}
+	logrus.Debugf("Using config file %s", configFile)
+
+	cfg, err := getConfig(configFile)
+	if err != nil {
+		logrus.WithError(err).Error("Failed to read config")
+		os.Exit(-1)
+	}
+	logrus.Debugf("Read config:\n%#v", cfg)
+
+	var pc PCClient
+	if err := pc.PackagecloudClient(
+		cfg.Packagecloud.Auth.User,
+		cfg.Packagecloud.Auth.Token,
+		cfg.Packagecloud.Repo); err != nil {
+		logrus.WithError(err).Error("Failed to create a Packagecloud client instance")
+		os.Exit(-1)
+	}
+	logrus.WithField("Packagecloud repo", pc.Repo).Info("Retrieving list of available files")
+
+	pcPackages, err := pc.PackagecloudList()
+	if err != nil {
+		logrus.WithFields(logrus.Fields{
```

Comment:
Ah - sorry, so you are using fields ;) In which case it would be good to be consistent and make all log calls structured like this.

Reply:
thanks!